### PR TITLE
Add dot_product example to documentation

### PR DESCRIPTION
**Pull request by spacewafer**
Proposed above is the example of a dot product arithmetic operation involving different point concepts as operands. This aims to aid the user in understanding the concept more clearly when the user goes through the documentation.

### References

- #658

### CODE REVIEW

File: doc/src/examples/dot_product [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+	
+	bg::model::point<double, 2, bg::cs::cartesian> point3(3.0, 2.0);
+	bg::model::point<double, 2, bg::cs::cartesian> point4(4.0, 7.0);
+	
+	dp2 = bg::dot_product(point3, point4);
+	
+ 	std::cout << "Dot Product 2: "<< dp2 << std::endl; 
+ 	
+ 	boost::array<double, 2> a =  {1, 2};
+  boost::array<double, 2> b =  {2, 3};
```
Comment (vissarion) [inline doc/src/examples/dot_product]: tab is missing

File: doc/src/examples/dot_product [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+    
+  dp3 = bg::dot_product(a, b);
```
Comment (vissarion) [inline doc/src/examples/dot_product]: tab

File: doc/src/examples/dot_product (hunk 1 above)
Comment (mloskot) [reply doc/src/examples/dot_product]: More precisely, 4 spaces as per https://github.com/boostorg/geometry/wiki/Guidelines-for-Developers#code-formatting-and-indentation

File: doc/src/examples/dot_product (hunk 2 above)
Comment (mloskot) [reply doc/src/examples/dot_product]: More precisely, 4 spaces as per https://github.com/boostorg/geometry/wiki/Guidelines-for-Developers#code-formatting-and-indentation

**Comment by spacewafer** [comment]
> It would be helpful if you could add to the commit message line
> 
> ```
> Closes #658
> ```

Can you please elaborate?

**Comment by mloskot** [comment]
@spacewafer I suggested to make the commit message to read

```
A dot_product example for the documentation

Closes #658
```

Then, GitHub will properly auto-link this PR with the issue this PR targets, it will also close the issue when the PR is merged. This is helpful when browsing the GitHub issues archive as well as when collecting release notes.

Nevermind, this can be done during "Squash and merge".

**Comment by awulkiew** [comment]
Something wierd is happening here. :)

**Comment by spacewafer** [comment]
> Something wierd is happening here. :)

Can you please elaborate on this?

**Comment by spacewafer** [comment]
> > Something wierd is happening here. :)
> 
> Can you please elaborate on this?

I hadn't added Doxygen commands before. Does this seem right now?

**Comment by vissarion** [comment]
Similarly to https://github.com/boostorg/geometry/pull/665 you have to verify that you can produce the documentation locally

File: doc/src/examples/dot_product [hunk 3]
```diff
@@ -0,0 +1,66 @@
+// Boost.Geometry
```
Comment (mloskot) [inline doc/src/examples/dot_product]: Please, remove this extension-less file `doc/src/examples/dot_product` which seems to be copy of the correct one `doc/src/examples/arithmetic/dot_product.cpp`.

**Pull request merged**